src/koheesio/categorical_encoding.py

```diff
@@ -0,0 +1,30 @@
+# Import necessary libraries
+from sklearn.preprocessing import OneHotEncoder
+import pandas as pd
+from pydantic import BaseModel
+
+# Define the EncodingConfig class
+class EncodingConfig(BaseModel):
+    drop_first: bool = True  # Whether to drop the first dummy column
+
+# Define the categorical_encoding function
+def categorical_encoding(data, columns, config: EncodingConfig):
+    """
+    Encodes categorical columns using one-hot encoding.
+
+    Args:
+        data (pd.DataFrame): Input dataset.
+        columns (list): List of categorical columns to encode.
+        config (EncodingConfig): Configuration for encoding.
+
+    Returns:
+        pd.DataFrame: Dataset with one-hot encoded columns.
+    """
```

Comment:
Docstrings are supposed to be in Numpy notation

Reply:
Modified Docstrings to be in Numpy notation